lib/wallet.py

```diff
@@ -1437,6 +1438,14 @@ def receive_history_callback(self, addr, hist, tx_fees):
         if self.network:
             self.network.trigger_callback('on_history', self)
 
+    def add_tx_to_history(self, txid):
+        with self.lock:
+            for addr in itertools.chain(list(self.txi.get(txid, {}).keys()), list(self.txo.get(txid, {}).keys())):
+                cur_hist = self._history.get(addr, list())
+                if not txid in itertools.chain(*cur_hist):
```

Comment:
Two things:

1. `not in` is considered more idiomatic in Python than `not x in y`
2. I think using `itertools` here is excessive -- maybe rewrite this as:

```python
if not any(True for x in cur_hist if x[0] == txid):
```
This way we avoid building the list of the second "height" item which is not needed...

Reply:
1. Sorry, forgot to enable flake8 (:
2. Because of list unpacking, right? itertools.chain itself returns an iterator, it doesn't build a new list

Reply:
Done